### PR TITLE
Improve my surveys component

### PR DESCRIPTION
**Pull request by EricThuaud**

### CODE REVIEW

File: src/ui/components/UserAccount/personalData/display.jsx [hunk 1]
```diff
@@ -3,18 +3,48 @@ import { formDictionary } from "i18n";
 
 export const PersonalDataDisplay = ({ user }) => {
   const { lastName, firstName, phone } = user;
-  const fonction= user.function;
+  const fonction = user.function;
```
Comment (laurentC35) [inline src/ui/components/UserAccount/personalData/display.jsx:6]: You can write
```js
const { lastName, firstName, phone, function: userFunction } = user
```

This allows to rename the problematic word "function" as "userFunction".

File: src/ui/shared/Header/index.jsx [hunk 2]
```diff
@@ -72,14 +77,21 @@ export const Header = () => {
...
-          <Link to="/portail">
-            <MenuItem>
+
+          <MenuItem
+            component={Link}
+            href={`${portailUrl}/aqv/contacter-assistance`}
+            target="_blank"
+            rel="noreferrer"
+          >
+            <Link>
```
Comment (laurentC35) [inline src/ui/shared/Header/index.jsx:87]: You can now remove the `<Link>` tag here.
It is useless now :-)

File: src/ui/components/SurveyList/item/index.jsx [hunk 3]
```diff
@@ -1,14 +1,111 @@
...
+
+        if (isPast(new Date(closingDate))) {
+          message = surveyDictionary.surveyMessageClosed(returnDate);
+        }
+        break;
+    }
+    return message;
+  };
+
+  const getLogoType = () => {
```
Comment (laurentC35) [inline src/ui/components/SurveyList/item/index.jsx:68]: Can you create a new component for the Logo to externalize this ?
You can create the Icon under "item" folder like `itemIcon.jsx`

File: src/ui/components/SurveyList/item/index.jsx [hunk 4]
```diff
@@ -1,14 +1,111 @@
...
+    if (
+      questioningStatus === HC_QUESTIONING ||
+      questioningStatus === REFUSAL_QUESTIONING ||
+      getSurveyStatus(openingDate, closingDate, returnDate).status === surveyDictionary.surveyClosed
+    ) {
+      return <CloseIcon />;
+    }
+    if (
+      getSurveyStatus((openingDate, closingDate, returnDate).status === surveyDictionary.surveyOpen) ||
+      getSurveyStatus((openingDate, closingDate, returnDate).status === surveyDictionary.surveyClosing)
```
Comment (laurentC35) [inline src/ui/components/SurveyList/item/index.jsx:86]: You call 4 times the function `getSurveyStatus` !
If you externalize the logo, you can comute the result only one time according to the parameters

File: src/ui/components/SurveyList/filter.jsx [hunk 5]
```diff
@@ -98,6 +177,39 @@ export const SmartFilter = ({ mySurveys, setSurveyFiltered, setPage }) => {
             ))}
           </Box>
         </Grid>
+        <Grid item xs={4}>
+          <FormControl fullWidth sx={{ backgroundColor: "white" }}>
+            <InputLabel id="demo-multiple-chip-label">{surveyDictionary.searchByStatus}</InputLabel>
```
Comment (laurentC35) [inline src/ui/components/SurveyList/filter.jsx:182]: You have the same id, I think you have to rename it (`id="demo-multiple-chip-label"`)

File: src/ui/components/SurveyList/filter.jsx [hunk 6]
```diff
@@ -12,12 +12,73 @@ import {
...
+  HC_QUESTIONING,
+  REFUSAL_QUESTIONING,
+} from "core/constants";
 
 export const SmartFilter = ({ mySurveys, setSurveyFiltered, setPage }) => {
   const [filter, setFilter] = useState("");
   const [selectedSurveysFilter, setSelectedSurveysFilter] = useState([]);
+  const [selectedStatusFilter, setSelectedStatusFilter] = useState([]);
+
+  const statusOrder = [
```
Comment (laurentC35) [inline src/ui/components/SurveyList/filter.jsx:29]: You can externalize this into core/constants in the file like `filterConstants.js`

File: src/ui/components/SurveyList/filter.jsx [hunk 7]
```diff
@@ -12,12 +12,73 @@ import {
...
+    surveyDictionary.surveyClosed,
+  ];
+  const lowOrderQuestioningStatus = [
+    VALINT_QUESTIONING,
+    VALPAP_QUESTIONING,
+    REFUSAL_QUESTIONING,
+    HC_QUESTIONING,
+  ];
 
+  const sortByQuestioningStatusBySurveyStatusByReturnDate = (a, b) => {
```
Comment (laurentC35) [inline src/ui/components/SurveyList/filter.jsx:48]: This function has to be externalize into `core/functions.js`

**Review by laurentC35 — CHANGES_REQUESTED** [review]
I have reviewed your code.
Well done !

But I have some requests for you.

File: src/ui/components/SurveyList/item/index.jsx [hunk 8]
```diff
@@ -1,14 +1,111 @@
...
+            variant="contained"
+            endIcon={<ContentPasteGo />}
+          >
+            <Typography>{surveyDictionary.accessSurvey}</Typography>
+          </Button>
+        </Link>
+      );
+    }
+
+    return "nothing";
```
[the remark this answers is not in the document]
Comment (EricThuaud) [reply src/ui/components/SurveyList/item/index.jsx:106]: yep^^ corrected

**Comment by laurentC35** [comment]
I have another detail.
Can you change the favicon.ico, logo192.png and logo512.png (under public folder) to the logo of marianne ?

- update `manifest.json file`  :

```json
{
"short_name" : "Portail Mes Enquêtes", 
"name" : "Portail de réponse aux enquêtes de la statistique publique",
...
}
```
- update `index.html` file

```xml
<head>
  ...
  <meta
      name="description"
      content="Portail de réponse aux enquêtes de la statistique publique" />
  ...
  </head>
```

File: src/ui/shared/Header/index.jsx (hunk 2 above)
Comment (EricThuaud) [reply src/ui/shared/Header/index.jsx:87]: done

File: src/core/functions.js [hunk 9]
```diff
@@ -10,12 +13,43 @@ export const chunkArray = (myArray, chunk_size) => {
...
+    toolTip = surveyDictionary.surveyOpenTooltip;
+  }
+  if (isPast(new Date(returnDate)) && isFuture(new Date(closingDate))) {
+    status = surveyDictionary.surveyClosing;
+    colorChip = "warning";
+    toolTip = surveyDictionary.surveyClosingTooltip;
+  }
+  if (isPast(new Date(closingDate))) {
+    status = surveyDictionary.surveyClosed;
+    colorChip = "error";
```
[the remark this answers is not in the document]
Comment (EricThuaud) [reply src/core/functions.js:38]: done

File: src/ui/components/SurveyList/filter.jsx (hunk 7 above)
Comment (EricThuaud) [reply src/ui/components/SurveyList/filter.jsx:48]: done

File: src/ui/components/SurveyList/item/index.jsx (hunk 3 above)
Comment (EricThuaud) [reply src/ui/components/SurveyList/item/index.jsx:68]: done, but keeping getLogoType function in itemIcon.jsx, returning getLogoType()

File: src/ui/components/SurveyList/item/index.jsx (hunk 4 above)
Comment (EricThuaud) [reply src/ui/components/SurveyList/item/index.jsx:86]: done

**Review by laurentC35 — APPROVED** [review]
Some improvements can be made.
But I think it's enough for now :)

Good work

**Pull request merged**